src/modules/governance/proposal/ProposalLifecycle.tsx

```diff
@@ -15,38 +11,114 @@ import {
   timelineItemClasses,
   TimelineSeparator,
 } from '@mui/lab';
-import { Button, Paper, SvgIcon, Typography, useTheme } from '@mui/material';
+import { Box, Button, IconButton, Paper, SvgIcon, Typography, useTheme } from '@mui/material';
 import dayjs from 'dayjs';
-import { ReactNode } from 'react';
+import { ReactNode, useState } from 'react';
+import { DarkTooltip } from 'src/components/infoTooltips/DarkTooltip';
 import { Link } from 'src/components/primitives/Link';
-import { EnhancedProposal } from 'src/hooks/governance/useProposal';
+import { Proposal } from 'src/hooks/governance/useProposals';
 import { useRootStore } from 'src/store/root';
+import { governanceV3Config } from 'src/ui-config/governanceConfig';
+import { getNetworkConfig } from 'src/utils/marketsAndNetworksConfig';
 import { GENERAL } from 'src/utils/mixPanelEvents';
 
-enum ProposalLifecycleStep {
-  Created,
-  OpenForVoting,
-  VotingClosed,
-  PayloadsExecuted,
-  Cancelled,
-  Expired,
-  Done,
-}
+import {
+  getLifecycleStateTimestamp,
+  getPayloadStateTimestamp,
+  getProposalStateTimestamp,
+  getVotingMachineProposalStateTimestamp,
+  ProposalLifecycleStep,
+} from '../utils/formatProposal';
 
-export const ProposalLifecycle = ({
-  proposal,
-  payloads,
-}: {
-  proposal: EnhancedProposal | undefined;
-  payloads: Payload[] | undefined;
-}) => {
+export const ProposalLifecycle = ({ proposal }: { proposal: Proposal | undefined }) => {
   const trackEvent = useRootStore((store) => store.trackEvent);
 
-  if (payloads === undefined || proposal === undefined) {
+  if (proposal === undefined) {
     return <></>; // TODO: skeleton
   }
 
-  const proposalState = getLifecycleState(proposal, payloads);
+  const createdPayloadOrder = proposal.payloadsData.sort((a, b) => a.createdAt - b.createdAt);
+
+  const coreNetworkConfig = getNetworkConfig(governanceV3Config.coreChainId);
+
+  const proposalCreatedSubsteps = createdPayloadOrder
+    .map<ProposalStepProps>((payload) => ({
+      completed: true,
```

Comment:
@JoaquinBattilana what does completed mean in this context

Reply:
It's a prop for the timeline component, so if its completed it shows the black dot